server/sonar-server/src/main/java/org/sonar/server/qualitygate/RegisterQualityGates.java

```diff
@@ -37,67 +48,205 @@
 
 public class RegisterQualityGates implements Startable {
 
-  private static final String BUILTIN_QUALITY_GATE = "SonarQube way";
+  private static final Logger LOGGER = Loggers.get(RegisterQualityGates.class);
+
+  private static final String BUILTIN_QUALITY_GATE_NAME = "Sonar way";
   private static final int LEAK_PERIOD = 1;
+  private static final String A_RATING = Integer.toString(RatingGrid.Rating.A.getIndex());
+
+  private static final List<QualityGateCondition> QUALITY_GATE_CONDITIONS = asList(
+    new QualityGateCondition().setMetricKey(NEW_SECURITY_RATING_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold(A_RATING),
+    new QualityGateCondition().setMetricKey(NEW_RELIABILITY_RATING_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold(A_RATING),
+    new QualityGateCondition().setMetricKey(NEW_MAINTAINABILITY_RATING_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold(A_RATING),
+    new QualityGateCondition().setMetricKey(NEW_COVERAGE_KEY).setOperator(OPERATOR_LESS_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold("80"),
+    new QualityGateCondition().setMetricKey(NEW_DUPLICATED_LINES_DENSITY_KEY).setOperator(OPERATOR_GREATER_THAN).setPeriod(LEAK_PERIOD).setErrorThreshold("3")
+  );
 
   private final DbClient dbClient;
   private final QualityGateConditionsUpdater qualityGateConditionsUpdater;
-  private final LoadedTemplateDao loadedTemplateDao;
   private final QualityGates qualityGates;
+  private final QualityGateDao qualityGateDao;
+  private final QualityGateConditionDao qualityGateConditionDao;
 
-  public RegisterQualityGates(DbClient dbClient, QualityGateConditionsUpdater qualityGateConditionsUpdater,
-    LoadedTemplateDao loadedTemplateDao, QualityGates qualityGates) {
+  public RegisterQualityGates(DbClient dbClient, QualityGateConditionsUpdater qualityGateConditionsUpdater, QualityGates qualityGates) {
     this.dbClient = dbClient;
     this.qualityGateConditionsUpdater = qualityGateConditionsUpdater;
-    this.loadedTemplateDao = loadedTemplateDao;
     this.qualityGates = qualityGates;
+    this.qualityGateDao = dbClient.qualityGateDao();
+    this.qualityGateConditionDao = dbClient.gateConditionDao();
   }
 
   @Override
   public void start() {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      if (shouldRegisterBuiltinQualityGate(dbSession)) {
-        createBuiltinQualityGate(dbSession);
-        registerBuiltinQualityGate(dbSession);
-        dbSession.commit();
+      QualityGateDto builtin = qualityGateDao.selectByName(dbSession, BUILTIN_QUALITY_GATE_NAME);
```

Comment:
I think we could use a different strategy : select all built-in profiles, check there's only one having name "Sonar way", and if not : 
- If only one but name is wrong => update name (and add a log)
- if many => update flags (and add a log)

Reply:
On this one, I tend to disagree. For the following reasons : 
* Doing this way will be slower (it's done at each startup), 
* Unless using order by it's not predictable,
* If for any reason the customer quality gate has the flag built-in, the conditions will be lost and will generate some misunderstanding 

I prefer to not automatically do magic things on customer data, with this strategy nothing is lost for the customer.

Reply:
ok it's fine for me